### ISSUE TITLE
Criteria Builder Like operator do not work on MongoDb.

### ISSUE THREAD

**Issue by cprashantgupta** [micronaut-projects/micronaut-data#2411]
### Expected Behavior

As per documented on : [micronaut pages](https://micronaut-projects.github.io/micronaut-data/latest/guide/#mongoCriteriaSpecifications) , I have created an implementation of PredicateSpecification<T>. The purpose of the implementation is to add criteria to query dynamically. 

```
public Predicate toPredicate(Root<Articles> root, CriteriaBuilder criteriaBuilder) {
        List<Predicate> predicates = new ArrayList<>();
        if (searchText != null && !searchText.isEmpty()) {
            String searchParameter = ".*" + searchText.trim() + "*.";
            Predicate predicateForBody = criteriaBuilder.like((root.get("documentText")), searchParameter);
            Predicate predicateForTitle = criteriaBuilder.like((root.get("title")), searchParameter);
            predicates.add(criteriaBuilder.or(predicateForBody, predicateForTitle));
        }
        if (status != null && !status.isEmpty()) {
            predicates.add(criteriaBuilder.equal(root.get("status"), StatusEnum.fromValue(status)));
        }
        return criteriaBuilder.and(predicates.toArray(new Predicate[]{}));
    }
```

I expect the query to work like contain, just the way it works if I use direct JPA query.

```
findAllByTenantIdAndParentIdIsNullAndTitleContainsOrDocumentTextContains
```

### Actual Behaviour

Mongo throws an error : 

```
02:19:14.325 [default-nioEventLoopGroup-1-3] ERROR i.m.http.server.RouteExecutor - Unexpected error occurred: Queries of type Like are not supported by this implementation
java.lang.IllegalArgumentException: Queries of type Like are not supported by this implementation
	at io.micronaut.data.document.model.query.builder.MongoQueryBuilder.handleCriterion(MongoQueryBuilder.java:909)
```

The query generated is have following : 
```
{"documentText": {"$eq": ".*Demo*."}}
```

Where as I want it to be 
```
{title : {$regex : "Demo"}}
```

### Steps To Reproduce

1. Create a mongo db micronaut project from micronaut lancher.
2. Connect to mongo database.
3. Implement the interface as follows 

```
import static jakarta.persistence.criteria.*;

public interface PredicateSpecification<T> {

    (1)
    @Nullable
    Predicate toPredicate(@NonNull Root<T> root, (2)
                          @NonNull CriteriaBuilder criteriaBuilder (3)
    );

}
```

### Environment Information

- Window
- Java 11
- Micronaut 3.9.4

### Version

3.9.4

**Comment by radovanradic** [comment]
Should be doable in next micronaut-data patch release

**Comment by cprashantgupta** [comment]
Thank you 👍 💯

**Comment by cprashantgupta** [comment]
@radovanradic  : I am trying to upgrade my micronaut application from 3.9.4 to 4.x due to above issue, but facing another issue : https://github.com/micronaut-projects/micronaut-maven-plugin/issues/805 . Can you please suggest what wrong with this ?

**Comment by radovanradic** [comment]
Sorry, not very familiar with that but I believe team working on it will handle it.

**Comment by cprashantgupta** [comment]
@radovanradic  : It looks the fix for like query is in master and not in 4.0.2 . So probably will be coming in 4.0.3. 

![image](https://github.com/micronaut-projects/micronaut-data/assets/43907358/cc56e3a6-811e-4cb5-abc8-2432119c40e4)

**Comment by radovanradic** [comment]
Right, the fix is not released yet and should be in 4.0.3.
